### PR TITLE
Add option to customize seek time

### PR DESCRIPTION
**Pull request by kabel2**
**Changes**
Adds a slider in the playback settings to set the fast-forward/rewind times. Minimum 1 second, maximum 60 seconds. Default 15 seconds.
TODO:
I think the naming of the setting needs to be changed. It may also be necessary to change the slider to two dropdowns (as in the Android smartphone app). 

**Issues**
Fixes my own frustration of not being able to rewind a short dialog.

### CODE REVIEW

**Comment by fice-t** [comment]
This should ideally take into account the user-specific settings made on the Jellyfin server, i.e. the `Skip {forward,back} length` values set via the web client's `menu->User->Settings`.

These settings are already partially used in the Android TV client, as [seen here](https://github.com/jellyfin/jellyfin-androidtv/blob/dcbf87488cd035a02e198e36eac7c39bcb04d446/app/src/main/java/org/jellyfin/androidtv/preference/UserSettingPreferences.kt#L17-L18) and in [this issue](https://github.com/jellyfin/jellyfin-androidtv/issues/3033).

Having an override specifically for the TV client can still make sense (a user may want different seeking behaviour between clients), but I think that it should still default to the server-side settings.

Also, perhaps the seek time should be cached somewhere so the preferences aren't being hit for each `getSeekPositions`? Could the seek time itself just be fed into `CustomSeekProvider`?

**Comment by kabel2** [comment]
I have already seen the settings for forward and back. However, the SeekProvider does not allow a distinction between forward and back, so there is only one value. 

The setting is therefore different from the Android app/web app. I think using only the value of “back” would not be much better than creating a new (local) value (seek time). Maybe this would be a temporary solution until there is a rewrite of the playback.

The seektime can of course be given directly to the CustomSeekProvider, would be no problem to change that.

**Comment by fice-t** [comment]
> However, the SeekProvider does not allow a distinction between forward and back, so there is only one value.

Right, that does complicate things. Luckily, the ATV playback rewrite [allows such a distinction](https://github.com/jellyfin/jellyfin-androidtv/blob/b12e4a69781bdb481a838d327cc977dd7be0fffc/playback/media3/session/src/main/kotlin/MediaSessionPlayer.kt#L132-L133).

Since there is the rewrite coming up, I would propose the following:

### Legacy code
- Use one local setting (`UserPreferences`)
  - Do not fall back to either server setting (`UserSettingPreferences`) to avoid possible confusion
  - Always enabled in the UI

### Rewrite code
- Use two local settings, one each for back/forward
  - Fall back to the respective server setting if not changed locally
  - Only enable each local setting in the UI if requested (perhaps gated by a checkbox?)
    - Either way, the UI should present each server-side value if the local setting is not enabled.

This PR already accomplishes the legacy part, and could be considered self-contained. I wrote some thoughts about how this PR could consider the transition between legacy/rewrite, but came to the conclusion that it doesn't matter much on the legacy side. Perhaps the names could be changed from `*seek_time*` -> `*seek_time_legacy*`, since in the above scheme the keys/descriptions would need to be different between legacy/rewrite. Also, maybe `duration` is better than `time` for the names?

So perhaps the rewrite side could be punted off for later. FWIW, I was previously planning on working on custom seek times once the rewrite was completed if no one else got to it first.

@nielsvanvelzen Thoughts on all this?

**Comment by nielsvanvelzen** [comment]
We should re-use the skip forward/backward preferences already available in the user settings. Not add a new one.

**Comment by nielsvanvelzen** [comment]
Hey @kabel2, are you still interested in finishing this pull request? it's currently waiting for you to address my comment above.

**Comment by kabel2** [comment]
Sorry for the late reply.
Without the rewrite of the player it is not possible to distinguish between forward and backwards. 
So should we use the mean value of forward and backwards or how should two settings be used for one value?

**Comment by nielsvanvelzen** [comment]
We can use the "forward" preference for now and update it later when we redo the player design.

**Comment by suchtie664** [comment]
Okay, I'll take a look at the necessary changes tomorrow :)

**Comment by kabel2** [comment]
Okay, I'll take a look at the necessary changes tomorrow :)

**Comment by kabel2** [comment]
Is now adjusted. 
But there is still a problem, the minimum skip length is just 0, although I have set it to `min = 5_000`.

File: app/src/main/java/org/jellyfin/androidtv/preference/UserSettingPreferences.kt [hunk 1]
```diff
@@ -15,7 +15,7 @@ class UserSettingPreferences(
 ) {
 	companion object {
 		val skipBackLength = intPreference("skipBackLength", 10_000)
-		val skipForwardLength = intPreference("skipForwardLength", 30_000)
+		val skipForwardLength = intPreference("skipForwardLength", 10_000)
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/preference/UserSettingPreferences.kt:18]: The default value needs to be consistent with jellyfin-web

https://github.com/jellyfin/jellyfin-web/blob/9fbc37fbf30e986451964d3cd8fd178d1a3895be/src/scripts/settings/userSettings.js#L366

File: app/src/main/java/org/jellyfin/androidtv/ui/playback/overlay/LeanbackOverlayFragment.java [hunk 2]
```diff
@@ -18,6 +20,7 @@ public class LeanbackOverlayFragment extends PlaybackSupportFragment {
     private VideoPlayerAdapter playerAdapter;
     private boolean shouldShowOverlay = true;
     private Lazy<PlaybackControllerContainer> playbackControllerContainer = inject(PlaybackControllerContainer.class);
+    private final Integer skipForwardLength = KoinJavaComponent.<UserSettingPreferences>get(UserSettingPreferences.class).get(UserSettingPreferences.Companion.getSkipForwardLength());
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/ui/playback/overlay/LeanbackOverlayFragment.java:23]: Move the reading of the setting to the actual usage and just inject the class here.

**Review by nielsvanvelzen — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/jellyfin/androidtv/preference/UserSettingPreferences.kt (hunk 1 above)
Comment (kabel2) [reply app/src/main/java/org/jellyfin/androidtv/preference/UserSettingPreferences.kt]: i changed the value to 30 seconds

File: app/src/main/java/org/jellyfin/androidtv/ui/playback/overlay/LeanbackOverlayFragment.java (hunk 2 above)
Comment (kabel2) [reply app/src/main/java/org/jellyfin/androidtv/ui/playback/overlay/LeanbackOverlayFragment.java]: I have created an extra variable for the skipForwardLength. Or should the whole UserSettingPreferences be passed to the CustomSeekProvider?

**Comment by nielsvanvelzen** [comment]
The tests are currently failing

**Review by nielsvanvelzen — APPROVED** [review]
I've made 2 final changes:

1. Fix the issue where you can change the value below the minimum - this wasn't your mistake but a bug in our UI code
2. Use the long type in CustomSeekProvider and do the casting from LeanbackOverlayFragment (where we initialize the provider)

Thanks for the pull request!

**Pull request merged**